fenix/app/src/main/java/org/mozilla/fenix/compose/ComposableFactory.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.tooling.preview.Preview
+import mozilla.components.concept.base.images.ImageLoadRequest
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.compose.ThumbnailImage as ThumbnailImageContent
+
+/**
+ * Factory object used to instantiate Composables with preview-breaking dependencies, such as
+ * [components].
+ */
+object ComposableFactory {
+
+    /**
+     * Loads a thumbnail belonging to a [ImageLoadRequest], fetching asynchronously the bitmap from storage.
+     * If this is in a preview, [fallbackContent] will be rendered instead.
+     *
+     * @param request [ImageLoadRequest] used to fetch the thumbnail bitmap.
+     * @param contentScale [ContentScale] used to draw image content.
+     * @param alignment [Alignment] used to draw the image content.
+     * @param modifier [Modifier] used to draw the image content.
+     * @param fallbackContent The content to display with a thumbnail is unable to be loaded.
+     */
+    @Composable
+    fun ThumbnailImage(
+        request: ImageLoadRequest,
+        contentScale: ContentScale,
+        alignment: Alignment,
+        modifier: Modifier = Modifier,
+        fallbackContent: @Composable () -> Unit,
+    ) {
+        if (inComposePreview) {
```

Comment:
we could optionally also create a compose wrapper for handling this logic check, if we felt like consolidating it further:

````
fun PreviewWrapper(
    previewContent: Lambda,
    actualContent: Lambda,
)